stores/directory/local.py

```diff
@@ -23,42 +26,81 @@ def __init__(self, local_dir: Union[str, Path]):
+        matching_files = [
+            f for f in self.get_file_names() if re.search(pattern, f)]
+
+        if len(matching_files) == 1:  # 1 file matched
+            return True
+        elif len(matching_files) == 0:  # 0 file matched
+            raise FileNotFoundError(f"No files found matching given pattern '{pattern}' in directory {self.local_path}")          
+        elif len(matching_files) > 1:  # 2+ files matched
+            raise FileNotFoundError(f"More than 1 file found that matches the regex pattern '{pattern}' in directory {self.local_path}")
+
+    def save_lone_file_matching(self, pattern: str, destination: Optional[Union[Path, str]] = None):
+        # Assert 1 file matches
+        if self.has_lone_file_matching(pattern):
+            file_name_list = self.get_file_names()
+            file_name = file_name_list[0]  
+            
+            if not destination:
+                open(file_name, "x")
+            else:
+                if isinstance(destination, str):
+                    save_path = Path(destination)
+                    save_path = save_path / file_name
+                    assert not save_path.exists(), f"Given file already exists in directory {save_path}."
+                    with (save_path).open("w") as saved_pattern:
+                        pass
+                else:
+                    save_path = destination / file_name
+                    assert not save_path.exists(), f"Given file already exists in directory {save_path}."
+                    with (save_path).open("w") as saved_pattern:
+                        pass
 
     def get_lone_matching_json_as_dict(self, pattern: str) -> dict:
-        #  To be implemented in local directory store 1 #5 
-        ...
+        # Assert 1 file matches
+        if self.has_lone_file_matching(pattern): 
+            file_names = self.get_file_names()
+            file_name = file_names[0]
+            file_path = self.local_path / file_names[0]
+            
+            if isinstance(file_name, str):
+                file_path = Path(file_path)
+
+            # use json.load to put contents of file into variable and return dict.
+            with open(file_path) as f:
+                json_dict = json.load(f)
+            return json_dict
 
     def save_lone_file_matching_regex(self, regex_str) -> Path:
         # Abstract methods can remain without implementation for now.
         ...
 
     def get_file_names(self) -> List[str]:
-        # Abstract methods can remain without implementation for now.
-        ...
-
-    def get_current_source_pathlike(self) -> str:
-        # Abstract methods can remain without implementation for now.
-        ...
-
-
-
-
+        file_names = os.listdir(self.local_path) #grab list of full paths to files,
+        #Check some files actually exist. 
+        if len(file_names) < 1:  # should this raise an error if empty or just fail silently and do nothing.. ?
```

Comment:
return an empty list, its feasible that someone wants to start from a point of an empty directory.

if file_names is length 0 and a list from above, you can just remove the conditional.